packages/common/src/hooks/usePurchaseMethod.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from 'react'
+
+import { USDC } from '@audius/fixed-decimal'
+import BN from 'bn.js'
+
+import { PurchaseMethod } from 'models/PurchaseContent'
+
+import { useUSDCBalance } from './useUSDCBalance'
+
+type UsePurchaseMethodProps = {
+  price: number
+  extraAmount: number | undefined
+  method: PurchaseMethod
+  setMethod: (value: PurchaseMethod) => void
+}
+
+/**
+ * Use existing balance as default payment method if available.
+ * Otherwise, default to card.
+ */
+export const usePurchaseMethod = ({
```

Comment:
I don't really get why this is a a hook ... or it should at least contain its own [method, setMethod] = useState rather than relying on those to be passed in

Reply:
don't we want to set the method for the entire formik form instead of local state tho?

Reply:
yeah i guess more of a readability/misdirection thing for me. i would do (pseudo code)

```
const hasSelectedMethod = !!method
const method = usePurchaseMethod(price, extraAmount, hasSelectedMethod)
useEffect(() => {
  setPurchaseMethod(purchaseMethod)
}, method]
```

It just seems less good to pass a form setter around to misc. hooks versus keeping it where the useField is used.

Or we wrap useField directly into the hook here

Reply:
no need to change here, just imo a little bit of a code smell